src/Payments/PaymentResult.php

```diff
@@ -0,0 +1,86 @@
+<?php
+/**
+ * Payment result.
+ *
+ * @package WooCommerce/Blocks
+ */
+
+namespace Automattic\WooCommerce\Blocks\Payments;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * PaymentResult class.
+ */
+class PaymentResult {
+	/**
+	 * List of valid payment statuses.
+	 *
+	 * @var array
+	 */
+	protected $valid_statuses = [ 'success', 'failure', 'pending', 'error' ];
```

Comment:
Are these defined somewhere in WooCommerce core as a list of payment statuses that payment methods can use? Can we reference that in the comment doc so if that changes we know it needs updated (or better yet, this is something that maybe could be added to WooCore at some point).

Reply:
`wc_get_order_statuses` provides that (with the `wc-` prefix), and plugins can extend into it new status.

Reply:
These are not core _order_ statuses. These are for payment status from the gateway. This is specific to the PaymentResult class and can be adjusted.

Reply:
To be clear, I'm asking if Woo Core has _payment_ statuses defined somewhere?  If so, we should probably pull them from where they are defined (if possible). If not, then we should make it really clear we're defining canonical statuses here (especially since these are functionally enums for the logic in this class).

Reply:
No they don't exist. The ajax response for payments in core is just success or fail. Nothing else.